python/lsst/ap/association/packageAlerts.py

```diff
@@ -0,0 +1,247 @@
+# This file is part of ap_association.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ("PackageAlertsConfig", "PackageAlertsTask")
+
+import os
+import tempfile
+
+import lsst.alert.packet as alertPack
+import lsst.geom as geom
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from lsst.utils import getPackageDir
+
+
+"""Methods for packaging Apdb and Pipelines data into Avro alerts.
+"""
+
+
+class PackageAlertsConfig(pexConfig.Config):
+    """Config class for AssociationTask.
+    """
+    schemaFile = pexConfig.Field(
+        dtype=str,
+        doc="Location to write alerts to.",
+        default=os.path.join(getPackageDir("sample_avro_alert"),
+                             "schema/2/1/lsst.alert.avsc"),
+    )
+    cutoutSize = pexConfig.RangeField(
+        dtype=int,
+        min=0,
+        max=1000,
+        default=30,
+        doc="Dimension of the square image cutouts to package in the alert."
+    )
+    alertWriteLocation = pexConfig.Field(
+        dtype=str,
+        doc="Location to write alerts to.",
+        default=os.path.join(os.getcwd(), "alerts"),
+    )
+
+
+class PackageAlertsTask(pipeBase.Task):
+    """Tasks for packaging Dia and Pipelines data into Avro alert packages.
+    """
+    ConfigClass = PackageAlertsConfig
+    _DefaultName = "packageAlerts"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.alertSchema = alertPack.Schema.from_file(self.config.schemaFile)
+        self.cutoutBBox = geom.Extent2I(self.config.cutoutSize,
+                                        self.config.cutoutSize)
+        os.makedirs(self.config.alertWriteLocation, exist_ok=True)
+
+    def run(self,
+            diaSourceCat,
+            diaObjectCat,
+            diaSrcHistory,
+            diffIm,
+            template,
+            ccdExposureIdBits):
```

Comment:
Depending on how the unique id discussion goes, we may be to have this. If we end up not using the ccdVisitId then this can leave. I'm okay leaving it for now and putting a TODO for when that ticket is resolved.

Reply:
Sounds good to me.